<a href="https://colab.research.google.com/github/SamaraMoreira/GS-EcoOcean/blob/main/GS_EcoOceanShorePollution.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# DataSet Coast Pollution Assessment

Link: https://www.kaggle.com/datasets/tzachymorad/shore-pollution-assessment


**Colunas:**


- Month: Representa o mês em que a amostra foi coletada.
- Season: Representa a estação do ano em que a amostra foi coletada.
- Shore: Indica a localização da amostra na costa.
- Pollution Level: Nível de poluição na amostra.
- Sample number: Número da amostra.
- Organic matter%: Percentual de matéria orgânica presente na amostra.
- Mean Number of Nematode species 1 per gram soil: Número médio de espécies de nematoides 1 por grama de solo.
- Mean Number of Turbillaria per gram soil: Número médio de Turbillaria por grama de solo.
- Mean Number of foraminefera per gram soil: Número médio de foraminíferos por grama de solo.
- Mean Number of Nematode species 2 per gram soil: Número médio de espécies de nematoides 2 por grama de solo.
- Water pH: pH da água.
- Soil pH: pH do solo.
- OC: Índice de carbono orgânico.
- Water Salinity: Salinidade da água.
- Soil Salinity: Salinidade do solo.
- P: Fósforo na amostra.
- Total dissolved solids: Sólidos totais dissolvidos na amostra.
- PP: Índice PP.
- Conduction: Condutividade elétrica.
- ORP: Potencial de oxirredução (ORP).
- Specific resistance: Resistência específica.
- Temp ©: Temperatura em Celsius.
- Conductivity: Condutividade.
- H: Hidrogênio na amostra.
- C-A: Índice C-A.
- C-B: Índice C-B.
- C-C: Índice C-C.

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np


In [2]:
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

In [3]:
ShorePollution = pd.read_csv('/content//Shore_Pollution.csv',encoding='latin1')

In [4]:
ShorePollution.head()

Month  Season  Shore  Pollution Level  Sample number  Organic matter%  \
0      8       4      1                0              1              NaN   
1      8       4      1                0              2              NaN   
2      8       4      1                0              3              NaN   
3      8       4      1                0              4              NaN   
4      8       4      1                0              5              NaN   

   Mean Number of Nematode species 1 per gram soil  \
0                                         5.700000   
1                                         3.288889   
2                                         8.444444   
3                                         1.866667   
4                                         3.377778   

   Mean Number of Turbillaria per gram soil  \
0                                  0.155556   
1                                  0.066667   
2                                  0.000000   
3                                  0.000000   
4                                  0.022222   

   Mean Number of foraminefera per gram soil  \
0                                        NaN   
1                                        NaN   
2                                        NaN   
3                                        NaN   
4                                        NaN   

   Mean Number of Nematode species 2 per gram soil  Water pH  Soil pH   OC  \
0                                              NaN     8.055      8.0  7.3   
1                                              NaN     8.055      8.0  7.3   
2                                              NaN     8.055      8.0  7.3   
3                                              NaN     8.055      8.0  7.3   
4                                              NaN     8.055      8.0  7.3   

   Water Salinity  Soil Salinity          P  Total dissolved solids      PP  \
0            38.5           11.0  28.396893                  57500.0  196.3   
1            38.5           11.0  28.396893                  57500.0  196.3   
2            38.5           11.0  28.396893                  57500.0  196.3   
3            38.5           11.0  28.396893                  57500.0  196.3   
4            38.5           11.0  28.396893                  57500.0  196.3   

   Conduction   ORP  Specific resistance   Temp ©  Conductivity   H  C-A  C-B  \
0     57400.0 -79.9                 17.42   30.12           NaN NaN  NaN  NaN   
1     57400.0 -79.9                 17.42   30.12           NaN NaN  NaN  NaN   
2     57400.0 -79.9                 17.42   30.12           NaN NaN  NaN  NaN   
3     57400.0 -79.9                 17.42   30.12           NaN NaN  NaN  NaN   
4     57400.0 -79.9                 17.42   30.12           NaN NaN  NaN  NaN   

   C-C  
0  NaN  
1  NaN  
2  NaN  
3  NaN  
4  NaN

In [6]:
print(ShorePollution.describe())

            Month      Season       Shore  Pollution Level  Sample number  \
count  219.000000  219.000000  219.000000       219.000000     219.000000   
mean     6.598174    2.260274    2.004566         1.009132       6.347032   
std      4.993867    1.023047    0.815547         0.818316       3.796199   
min      1.000000    1.000000    1.000000         0.000000       1.000000   
25%      3.000000    1.000000    1.000000         0.000000       3.000000   
50%      5.000000    2.000000    2.000000         1.000000       6.000000   
75%      9.000000    3.000000    3.000000         2.000000       9.000000   
max     18.000000    4.000000    3.000000         2.000000      14.000000   

       Organic matter%  Mean Number of Nematode species 1 per gram soil  \
count       202.000000                                       219.000000   
mean          0.453037                                         6.366831   
std           0.228357                                         6.078124   
min   

Analisando esses dados estatísticos, podemos obter várias informações úteis e fazer algumas observações interessantes:

1. **Período e Amostragem**:
   - **Número de amostras**: Em média, há aproximadamente 6 amostras por mês, com um desvio padrão de 3.79, indicando variabilidade na coleta de amostras.

2. **Estação do Ano**:
   - **Estação**: Os valores variam de 1 a 4, provavelmente correspondendo às quatro estações do ano, com uma distribuição bastante uniforme.
   
3. **Poluição e Ambientes de Costa**:
   - **Costa**: Três categorias, com uma média ligeiramente acima de 2.
   - **Nível de Poluição**: Escala de 0 a 2, com uma média de aproximadamente 1, sugerindo uma distribuição equilibrada entre áreas mais e menos poluídas.

4. **Matéria Orgânica e Espécies**:
   - **% de Matéria Orgânica**: Varia de 0.12% a 1.87%, com uma média de 0.45%.
   - **Número Médio de Espécies de Nematódeos 1**: Varia de 0 a 25, com uma média de aproximadamente 6.37.
   - **Número Médio de Turbellaria**: Relativamente baixo, com uma média de 0.17.
   - **Número Médio de Foraminíferos**: Varia amplamente de 0 a 73.60, com uma média de 15.07.
   - **Número Médio de Espécies de Nematódeos 2**: Média de 6.91, com um valor máximo de quase 39.

5. **Parâmetros de Solo e Água**:
   - **pH da Água**: Levemente alcalino, com uma média de 8.18.
   - **pH do Solo**: Média de 8.24, também levemente alcalino.
   - **Salinidade da Água e do Solo**: Valores médios de 37.35 e 9.83, respectivamente.
   - **Condução e Resistência Específica**: Média de 56280 µS/cm para condução e 17.98 ohm/cm para resistência.
   - **TDS (Total de Sólidos Dissolvidos)**: Média de 56036 mg/L.

6. **Parâmetros Adicionais**:
   - **ORP (Potencial de Oxidação-Redução)**: Média de -79.27, indicando condições redutoras.
   - **Temperatura da Água**: Varia de 14.7°C a 31.18°C, com uma média de 23.88°C.
   - **Condutividade e Concentrações de Elementos (C-A, C-B, C-C)**: Indicam variabilidade na composição química do ambiente.

### Observações:

1. **Distribuição Temporal**:
   - A variabilidade nas amostras por mês e o período estendido dos dados sugerem um esforço contínuo e longo de coleta de dados, possibilitando a análise de tendências sazonais e anuais.

2. **Poluição e Saúde do Ecossistema**:
   - A média de níveis de poluição em torno de 1 indica que há áreas com diferentes níveis de poluição, permitindo estudos comparativos sobre o impacto da poluição nos parâmetros ecológicos.

3. **Diversidade Biológica**:
   - A variabilidade nos números de espécies de nematódeos e foraminíferos sugere uma diversidade biológica significativa, que pode ser correlacionada com os níveis de poluição e outros parâmetros ambientais.

4. **Qualidade do Solo e da Água**:
   - O pH ligeiramente alcalino tanto da água quanto do solo pode afetar a disponibilidade de nutrientes e a biota. A alta condutividade e TDS indicam alta mineralização, típica de ambientes costeiros.

In [7]:
colunasShorePollution = ShorePollution.columns.values

In [8]:
colunasShorePollution

array(['Month', 'Season', 'Shore', 'Pollution Level', 'Sample number',
       'Organic matter%',
       'Mean Number of Nematode species 1 per gram soil',
       'Mean Number of Turbillaria per gram soil',
       'Mean Number of foraminefera per gram soil',
       'Mean Number of Nematode species 2 per gram soil', 'Water pH',
       'Soil pH', 'OC', 'Water Salinity', 'Soil Salinity', 'P',
       'Total dissolved solids ', 'PP', 'Conduction', 'ORP',
       'Specific resistance ', 'Temp ©', 'Conductivity', 'H', 'C-A',
       'C-B', 'C-C'], dtype=object)

# Refatorando o DataSet
Visando uma futura junção de datasets, realizei a alteração do nome da coluna Shore para Country, ambas colunas possuindo dados que varia de 1, 2 e 3. Além de apagar colunas que julguei pouco importante para a ánalise

In [9]:
NewShorePollution = ShorePollution.rename(columns={'Shore': 'Country'})

# Excluir as colunas 'H', 'C-A', 'C-B', 'C-C'
colunas_para_excluir = ['H', 'C-A', 'C-B', 'C-C']
NewShorePollution = NewShorePollution.drop(columns=colunas_para_excluir)

# Exibir o novo DataFrame
print(NewShorePollution.head())

# Se quiser salvar o novo DataFrame em um arquivo CSV
NewShorePollution.to_csv('NewShorePollution.csv', index=False)

   Month  Season  Country  Pollution Level  Sample number  Organic matter%  \
0      8       4        1                0              1              NaN   
1      8       4        1                0              2              NaN   
2      8       4        1                0              3              NaN   
3      8       4        1                0              4              NaN   
4      8       4        1                0              5              NaN   

   Mean Number of Nematode species 1 per gram soil  \
0                                         5.700000   
1                                         3.288889   
2                                         8.444444   
3                                         1.866667   
4                                         3.377778   

   Mean Number of Turbillaria per gram soil  \
0                                  0.155556   
1                                  0.066667   
2                                  0.000000   
3                 